### PR TITLE
Generate .hex file from .elf - closes #6

### LINKED ISSUE #6: Always build .bin & .hex
With https://github.com/ARMmbed/mbed-os/pull/13011, mbed-os tools now build both `.bin` and `.hex`.

I've added the following to `MBedExecutable.cmake`:

```cmake
  set(HEX_FILE ${CMAKE_CURRENT_BINARY_DIR}/${EXECUTABLE}.hex)
  # [...]
  # generate .hex firmware file
	add_custom_command(
		TARGET ${EXECUTABLE} POST_BUILD
		COMMAND arm-none-eabi-objcopy -O ihex $<TARGET_FILE:${EXECUTABLE}> ${HEX_FILE}
		WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
		COMMENT "Objcopying ${EXECUTABLE} to mbed firmware ${HEX_FILE}")
```

It works, but I'm wondering if it is the best way to do it? If so, I can provided a PR. :)

### PR DESCRIPTION
**Pull request by ladislas**

### CODE REVIEW

File: cmake/MBedExecutable.cmake [hunk 1]
```diff
@@ -46,34 +46,45 @@ function(add_mbed_executable EXECUTABLE)
...
 		set(MAP_FILE ${CMAKE_CURRENT_BINARY_DIR}/${EXECUTABLE}${CMAKE_EXECUTABLE_SUFFIX}.map)
 
-		set(OBJCOPY_COMMAND ${OBJCOPY_EXECUTABLE} --bin -o ${BIN_FILE} $<TARGET_FILE:${EXECUTABLE}>)
+		set(OBJCOPY_ELF_TO_BIN_COMMAND ${OBJCOPY_EXECUTABLE} --bin -o ${BIN_FILE} $<TARGET_FILE:${EXECUTABLE}>)
+		set(OBJCOPY_ELF_TO_HEX_COMMAND ${OBJCOPY_EXECUTABLE} --i32 -o ${HEX_FILE} $<TARGET_FILE:${EXECUTABLE}>)
 
 	elseif("${MBED_TOOLCHAIN_NAME}" STREQUAL "GCC_ARM")
 
 		set(MAP_FILE ${CMAKE_CURRENT_BINARY_DIR}/${EXECUTABLE}.map)
+		set(HEX_FILE ${CMAKE_CURRENT_BINARY_DIR}/${EXECUTABLE}.hex)
```
Comment (multiplemonomials) [inline cmake/MBedExecutable.cmake]: I don't think this line is doing anything since it's already in the block above.

**Review by multiplemonomials — COMMENTED** [review]
Looks good, just one thing to fix.

File: cmake/MBedExecutable.cmake (hunk 1 above)
Comment (ladislas) [reply cmake/MBedExecutable.cmake]: you're right! I've removed the duplicate.

**Comment by ladislas** [comment]
I've rebased the changes on master.

**Comment by ladislas** [comment]
Could this be merged or do you need me to modify more things?

**Comment by multiplemonomials** [comment]
Finally merged!  Sorry I had to block out some time to test both on GCC and Arm Compiler.

**Comment by ladislas** [comment]
Thanks a lot!

**Pull request merged**